File: solutions/java/acronym/1/src/main/java/Acronym.java
class Acronym {
    private final String acronym;

    Acronym(String phrase) {
        // 1. Remove all punctuation except letters, spaces, and hyphens
        String cleaned = phrase.replaceAll("[^a-zA-Z\\s-]", "");
        // 2. Treat hyphens as word separators (replace with space)
        String spaced = cleaned.replace('-', ' ');
        // 3. Split by one or more whitespace characters
        String[] words = spaced.trim().split("\\s+");

        StringBuilder sb = new StringBuilder();
        for (String word : words) {
            if (!word.isEmpty()) {
                // Take the first character of each word and uppercase it
                sb.append(Character.toUpperCase(word.charAt(0)));
            }
        }
        this.acronym = sb.toString();
    }

    String get() {
        return acronym;
    }
}

File: solutions/r/acronym/1/acronym.R
library(stringr)

acronym <- function(input) {
  # Replace hyphens with spaces (hyphens are word separators)
  input <- str_replace_all(input, "-", " ")
  
  # Remove all non-alphabetic characters except spaces
  input <- str_replace_all(input, "[^a-zA-Z ]", "")
  
  # Split into words, take first letter of each, convert to uppercase
  words <- str_split(input, "\\s+")[[1]]
  words <- words[words != ""]  # Remove empty strings
  
  # Get the first letter of each word and collapse into a single string
  paste0(toupper(str_sub(words, 1, 1)), collapse = "")
}